src/domain/event/eventHierarchy/__tests__/EventHierarchy.test.tsx

```diff
@@ -0,0 +1,217 @@
+const subSubEvents = fakeEvents(
+  subSubEventFields.length,
+  subSubEventFields.map(({ id, name }) => ({
+    id,
+    name: { fi: name },
+    subEvents: subSubSubEvents.data,
+    superEventType: SuperEventType.Recurring,
+  }))
+);
+
+const subSubPage2Events = fakeEvents(
+  subSubEventPage2Fields.length,
+  subSubEventPage2Fields.map(({ id, name }) => ({ id, name: { fi: name } }))
+);
+
+const subEvents = fakeEvents(
+  subEventFields.length,
+  subEventFields.map(({ id, name }) => ({
+    id,
+    name: { fi: name },
+    subEvents: subSubEvents.data,
+    superEventType: SuperEventType.Umbrella,
+  }))
+);
+
+const event = fakeEvent({
+  id: eventId,
+  name: { fi: eventName },
+  subEvents: subEvents.data,
+  superEventType: SuperEventType.Umbrella,
+});
+
+const count = subSubEventFields.length + subSubEventPage2Fields.length;
+const subSubEventsResponse = {
+  data: {
+    events: {
+      ...subSubEvents,
+      meta: {
+        ...subSubEvents.meta,
+        count,
+        next: 'http://localhost:8000/v1/event/?page=2',
+      },
+    },
+  },
+};
+const subSubEventsPage2Response = {
+  data: {
+    events: {
+      ...subSubPage2Events,
+      meta: {
+        ...subSubEvents.meta,
+        count,
+        previous: 'http://localhost:8000/v1/event/',
+      },
+    },
+  },
+};
+
+const subSubSubEventsResponse = {
+  data: {
+    events: subSubSubEvents,
+  },
+};
+const baseVariables = {
+  createPath: undefined,
+  include: ['audience', 'keywords', 'location', 'sub_events', 'super_event'],
+  pageSize: 100,
+  showAll: true,
+  sort: 'start_time',
+};
+const subSubEventsVariables = {
+  ...baseVariables,
+  superEvent: subEventFields[0].id,
+};
+const subSubSubEventsVariables = {
+  ...baseVariables,
+  superEvent: subSubEventFields[0].id,
+};
+
+const mocks = [
+  {
+    request: {
+      query: EventsDocument,
+      variables: subSubEventsVariables,
+    },
+    result: subSubEventsResponse,
+  },
+  {
+    request: {
+      query: EventsDocument,
+      variables: { ...subSubEventsVariables, page: 2 },
+    },
+    result: subSubEventsPage2Response,
+  },
+  {
+    request: {
+      query: EventsDocument,
+      variables: subSubSubEventsVariables,
+    },
+    result: subSubSubEventsResponse,
+  },
+];
+
+const renderComponent = () =>
+  render(<EventHierarchy event={event} />, { mocks });
+
+const eventsShouldBeVisible = async (eventNames: string[]) => {
+  for (const name of eventNames) {
+    await screen.findByText(name);
+  }
+};
+
+const eventsShouldBeHidden = async (eventNames: string[]) => {
+  for (const name of eventNames) {
+    await expect(screen.queryByText(name)).not.toBeInTheDocument();
+  }
+};
+
+test('should render all events in heirarchy by default', async () => {
+  renderComponent();
+  const allEvents = [
+    eventName,
+    ...flatMap(subEventFields, 'name'),
+    ...flatMap(subSubEventFields, 'name'),
+    ...flatMap(subSubEventPage2Fields, 'name'),
+    ...flatMap(subSubSubEventFields, 'name'),
```

Comment:
I'm not quite sure why flatMap is used here. Map would work also?

Reply:
I don't event remember why I used flatMap here. In some cases there might be some performance benefits, but not here. Replaced flatMap with map in all places where used